src/test/statistics/chunk_statistics/histograms/equal_distinct_count_histogram_test.cpp

```diff
@@ -0,0 +1,765 @@
+#include <limits>
+#include <memory>
+#include <string>
+
+#include "base_test.hpp"
+#include "gtest/gtest.h"
+
+#include "statistics/chunk_statistics/histograms/equal_distinct_count_histogram.hpp"
+#include "statistics/chunk_statistics/histograms/histogram_utils.hpp"
+#include "utils/load_table.hpp"
+
+namespace opossum {
+
+class EqualDistinctCountHistogramTest : public BaseTest {
+  void SetUp() override {
+    _int_float4 = load_table("src/test/tables/int_float4.tbl");
```

Comment:
is using these extremely simple tables really sufficient to test the cardinality estimation of the histograms? Wouldn't bins specified by hand have been better, at least for the numerical histograms?

Reply:
I think they are sufficient, because as long as there are multiple bins and multiple values in those bins (except for tests for those special cases) the tests should be the same.
Since the cardinality is a float and not an int, if the math was not correct, you would see it pretty quickly with testing cardinalities close to the bins, which pretty much every test does.